app.js

```diff
@@ -310,16 +312,31 @@ module.exports = {
         return newClient(srcUrl, opts);
       })
       // Validate the DB exists, before proceeding to backup
-      .then((backupDbClient) => validateBackupDb(backupDbClient))
-      .then((backupDbClient) => {
-        // If resuming write a newline as it's possible one would be missing from
-        // an interruption of the previous backup. If the backup was clean this
-        // will cause an empty line that will be gracefully handled by the restore.
-        if (opts.mode === 'full' && opts.resume) { // resume is valid in full mode only
-          targetStream.write('\n');
+      .then(backupDbClient => validateBackupDb(backupDbClient))
+      .then(backupDbClient => {
+        // Write either a file header or a resume marker.
+        let metadataToWrite;
+        if (opts.mode === 'full' && opts.resume) {
+          // resume is valid in full mode only
+          // Write the resume marker and a newline as it's possible one would be missing from
+          // an interruption of the previous backup. If the backup was clean this
+          // will cause an empty line that will be gracefully handled by the restore.
+          debug('Will write resume marker.');
+          metadataToWrite = `${RESUME_COMMENT}\n`;
+        } else {
+          // Write a file header including the name, version and mode
+          debug('Will write backup file header.');
+          metadataToWrite = `${JSON.stringify({ name: pkg.name, version: pkg.version, mode: opts.mode })}\n`;
         }
-        return backup(backupDbClient, opts, targetStream, ee);
+        return new Promise((resolve, reject) => {
+          targetStream.write(metadataToWrite, 'utf-8', (err) => {
+            if (err) { reject(err); } else { resolve(backupDbClient); }
```

Comment:
Oh, this use of resolve is neat.

Reply:
I think I've played with too many PassThrough streams now, so pass through promises!